.github/workflows/prereq.yml

```diff
@@ -18,13 +20,24 @@ jobs:
           fetch-depth: 500
       - name: Environment
         run: |
-          echo $GITHUB_REF_NAME
-          git rev-list -1 HEAD
-          git rev-list -1 HEAD^1
-          git rev-list -1 HEAD^2
+          echo "GITHUB_REF_NAME: $GITHUB_REF_NAME"
+          echo "GITHUB_BASE_REF: $GITHUB_BASE_REF"
+          echo "HEAD: $(git rev-list -1 HEAD)"
+          if test -n "$GITHUB_BASE_REF"; then
+            BASEB=$(git rev-list -1 HEAD^1)
+            HEADB=$(git rev-list -1 HEAD^2)
+          else
+            git fetch --no-tags --no-recurse-submodules --depth=500 origin main
+            BASEB=origin/main
```

Comment:
Also, this has a problem I'm not sure how to address.
If my github main branch is super old (because I didn't push the main ref, only my branch's ref),
then this gives crazy results sometimes (though enough bad things happen and are caught by the 23 bad email addresses we have in the repo today).
Also, we only fetch main's refs, not the branches...

Reply:
Yes, a tricky question. I personally do not see an optimal solution for now. Probably, this additional option can be considered: https://reviews.freebsd.org/D40261#916935